components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/storage/SyncedTabsStorage.kt

```diff
@@ -23,30 +27,42 @@ import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
 /**
  * A storage that listens to the [BrowserStore] changes to synchronize the local tabs state
  * with [RemoteTabsStorage].
+ *
+ * @param accountManager Account manager used to retrieve synced tabs.
+ * @param store Browser store to observe for state changes.
+ * @param tabsStorage Storage layer for tabs to sync.
+ * @param debounceMillis Length to debounce rapid changes for storing.
+ * @param onStoreComplete Callback for action to take after tabs are stored.
  */
 class SyncedTabsStorage(
     private val accountManager: FxaAccountManager,
     private val store: BrowserStore,
-    private val tabsStorage: RemoteTabsStorage = RemoteTabsStorage()
+    private val tabsStorage: RemoteTabsStorage = RemoteTabsStorage(),
+    private val debounceMillis: Long = 1000L,
+    private val onStoreComplete: () -> Unit = {}
 ) : SyncedTabsProvider {
     private var scope: CoroutineScope? = null
 
     /**
      * Start listening to browser store changes.
      */
+    @OptIn(FlowPreview::class)
     fun start() {
         scope = store.flowScoped { flow ->
-            flow.ifChanged { it.tabs }.map { state ->
+            flow.ifChanged { it.toSyncTabState() }.map { state ->
                 // TO-DO: https://github.com/mozilla-mobile/android-components/issues/5179
                 val iconUrl = null
-                state.tabs.filter { !it.content.private }.map { tab ->
+                state.tabs.filter { !it.content.private && !it.content.loading }.map { tab ->
                     // TO-DO: https://github.com/mozilla-mobile/android-components/issues/1340
                     val history = listOf(TabEntry(tab.content.title, tab.content.url, iconUrl))
                     Tab(history, 0, tab.lastAccess)
                 }
-            }.collect { tabs ->
-                tabsStorage.store(tabs)
             }
+                .debounce(debounceMillis)
```

Comment:
super minor nit only if you're updating your PR for other reasons:

I didn't realize the `map` was on the same line (and the linter doesn't complain) so the braces are super weird and I can't un-see it. 🙈 

Could you move it to the next line and reformat that block to be tabbed up?

```kotlin
            flow.ifChanged { it.toSyncTabState() }
                .map { state ->
                    // TO-DO: https://github.com/mozilla-mobile/android-components/issues/5179
                    val iconUrl = null
                    state.tabs.filter { !it.content.private && !it.content.loading }.map { tab ->
                        // TO-DO: https://github.com/mozilla-mobile/android-components/issues/1340
                        val history = listOf(TabEntry(tab.content.title, tab.content.url, iconUrl))
                        Tab(history, 0, tab.lastAccess)
                    }
                }
                .debounce(debounceMillis)

```

Reply:
Oh yeah, that was super gross. Thanks for calling it out!